tests/reconstruction/test_ssb.py

```diff
@@ -1,228 +1,625 @@
-    # spatial freq. step size according to probe raster
-    d_Qp = 1/dpix/Nblock
-
-    result_f = np.zeros(data.shape[:2], dtype=rearranged_ffts.dtype)
-
-    masks = np.zeros_like(data)
-
-    if cx is None:
-        cx = data.shape[-1] / 2
-    if cy is None:
-        cy = data.shape[-2] / 2
-
-    y, x = np.ogrid[0:Nscatter[0], 0:Nscatter[1]]
-    filter_center = (y - cy)**2 + (x - cx)**2 < semiconv_pix**2
-
-    for q in range(Nblock[0]):
-        for p in range(Nblock[1]):
-            qp = np.array((q, p))
-            flip = qp > Nblock / 2
-            real_qp = qp.copy()
-            real_qp[flip] = qp[flip] - Nblock[flip]
-            sx, sy = real_qp * d_Qp / d_Kf
-
-            filter_positive = (y - cy - sy)**2 + (x - cx - sx)**2 < semiconv_pix**2
-            filter_negative = (y - cy + sy)**2 + (x - cx + sx)**2 < semiconv_pix**2
-
-            mask_positive = np.all((filter_center, filter_positive,
-                                    np.invert(filter_negative)), axis=0)
-            mask_negative = np.all((filter_center, filter_negative,
-                                    np.invert(filter_positive)), axis=0)
-
-            f = rearranged_ffts[q, p]
-
-            non_zero_positive = np.count_nonzero(mask_positive)
-            non_zero_negative = np.count_nonzero(mask_negative)
-
-            if non_zero_positive > 0 and non_zero_negative > 0:
-                result_f[q, p] = (np.average(f[mask_positive]) - np.average(f[mask_negative])) / 2
-                masks[q, p] = ((mask_positive / non_zero_positive) - (
-                               mask_negative / non_zero_negative)) / 2
-            else:
-                assert non_zero_positive == 0
-                assert non_zero_negative == 0
-
-    result_f[0, 0] = np.average(rearranged_ffts[0, 0, filter_center])
-    # generate_masks doesn't patch the 0,0 mask (yet)
-
-    return result_f, masks, filter_center
+from os.path import isfile
+
+import pytest
+import numpy as np
+from scipy.io import loadmat
+import json
+
+from libertem import api as lt
+from libertem.executor.inline import InlineJobExecutor
+from libertem.io.dataset.memory import MemoryDataSet
+from libertem.masks import circular
+from libertem.corrections.coordinates import identity, rotate_deg
+from libertem.common.container import MaskContainer
+
+from ptychography.reconstruction.ssb import SSB_UDF, generate_masks, mask_tile_pair, get_results
+from ptychography.reconstruction.common import wavelength
+
+try:
+    from libertem.utils.devices import detect, has_cupy
+    from libertem.common.backend import set_use_cpu, set_use_cuda
+    use_cupy = detect()['cudas'] and has_cupy()
+except ImportError:
+    use_cupy = False
+
+DATA_PATH = "/storage/holo/clausen/testdata/ER-C-1/projects/ptycho-4.0/data/RefData/slice_00002_thick_0.312293_nm_blocksz.mat"
+PARAM_PATH = "/storage/holo/clausen/testdata/ER-C-1/projects/ptycho-4.0/data/slice_00002_thick_0.312293_nm_blocksz.params.json"
```

Comment:
@uellue 
Is there a way, that you can provide the test data? I want to test the implementation.

Reply:
@SimeonEhrig no problem! Does this link work? https://iffcloud.fz-juelich.de/s/ee72R8D9Rkbfrez

Reply:
We are preparing a better way to share this, perhaps Zenodo upload